src/context/ContextTypes.ts

```diff
@@ -194,21 +293,14 @@ export interface Valuation {
   type: string;
   valuationTime?: Date;
   value: number;
-  id?: { [key: string]: string };
+  id?: { [key: string]: any };
```

Comment:
So we are okay with the `any` here? Not a fan, as you know.

Reply:
If we're not we can't use quicktype + Typescript anymore. Now that the indexSignature is inherited by subtypes (specific context types like Instrument), it conflicts with their option properties and the generated types don't compile (the same occurs if the subtype expresses `additionalProperties` instead, as it generates the same index signature). Without the index signature the type can't be extended (due to excess property checking). 

This limitation *only* affects the TypeScrpt types. Specified id types remain typed (so id.BBG has to be a string) but if you made one up its allowed to be `any` by the typescript types generated (so `id.madeUpId = true` ends up valid according to Typescript) - however the JSONSchema (which actually handles the `unevaluatedProperties` - where quicktype ignores it/its not expressible in Typescript easily) would mark that as invalid.

In a hand produced type we could do this:

`id?: { [key: "BBG" | "ticker" | string]: string };`  (filling in all the defined ID types)

However, I'm not sure we can express that in JSONSchema such that it will generate as additionalProperties only affects values not the keys (the key type comes from the JSON spec, I think, where they are always required to be strings). I'll have a poke at the JSONschema docs to double check that's not possible - but don't think it is. Unless I turn up something, I think this proposal is the best we can do.

Reply:
patternProperties could have a regex defining the `id` field names, but also doesn't generate typescript. JSONSchema doesn't have an undefined type so we can't use that to make the index signature:

`id?: { [key: string]: string | undefined };`

which would not conflict with the optional properties. 

Hence, I don't see another way forward other than:
- accept the change in this PR which allows *unstandardized* id field properties to be `any` (standardized ones are still `string`)
- contribute a modification to quicktype causing it to generate `id?: { [key: string]: string | undefined };` when `additionalProperties: {type: string}` is expressed in a schema, and work with a fork until they merge it. I'm not sure this would be correct in all cases and hence whether they would accept it... 
- continue with out of date quicktype with vulnerable dependencies and types that can't be extended.